gms/impl/src/main/java/com/linkedin/metadata/resources/dataprocess/DataProcesses.java

```diff
@@ -0,0 +1,167 @@
+                .setName(processSnapshot.getUrn().getNameEntity())
+                .setOrigin(processSnapshot.getUrn().getOriginEntity());
+        ModelUtils.getAspectsFromSnapshot(processSnapshot).forEach(aspect -> {
+            if (aspect instanceof DataProcessInfo) {
+                DataProcessInfo processInfo = DataProcessInfo.class.cast(aspect);
+                value.setDataProcessInfo(processInfo);
+            }
+        });
+
+        return value;
+    }
+
+    @Nonnull
+    private DataProcessInfo getDataProcessInfoAspect(@Nonnull DataProcess process) {
+        final DataProcessInfo processInfo = new DataProcessInfo();
+        if (process.getDataProcessInfo().hasInputs()) {
+            processInfo.setInputs(process.getDataProcessInfo().getInputs());
+        }
+        if (process.getDataProcessInfo().hasOutputs()) {
+            processInfo.setOutputs(process.getDataProcessInfo().getOutputs());
+        }
+        return processInfo;
+    }
+
+    @Nonnull
+    @Override
+    protected DataProcessSnapshot toSnapshot(@Nonnull DataProcess process, @Nonnull DataProcessUrn urn) {
+        final List<DataProcessAspect> aspects = new ArrayList<>();
+        aspects.add(ModelUtils.newAspectUnion(DataProcessAspect.class, getDataProcessInfoAspect(process)));
+        return ModelUtils.newSnapshot(DataProcessSnapshot.class, urn, aspects);
+    }
+
+    @RestMethod.Get
+    @Override
+    @Nonnull
+    public Task<DataProcess> get(@Nonnull ComplexResourceKey<DataProcessKey, EmptyRecord> key,
+                                 @QueryParam(PARAM_ASPECTS) @Optional("[]") String[] aspectNames) {
+        return super.get(key, aspectNames);
+    }
+
+    @RestMethod.BatchGet
+    @Override
+    @Nonnull
+    public Task<Map<ComplexResourceKey<DataProcessKey, EmptyRecord>, DataProcess>> batchGet(
+            @Nonnull Set<ComplexResourceKey<DataProcessKey, EmptyRecord>> keys,
+            @QueryParam(PARAM_ASPECTS) @Optional("[]") String[] aspectNames) {
+        return super.batchGet(keys, aspectNames);
+    }
+
+    @Action(name = ACTION_INGEST)
+    @Override
+    @Nonnull
+    public Task<Void> ingest(@ActionParam(PARAM_SNAPSHOT) @Nonnull DataProcessSnapshot snapshot) {
+        return super.ingest(snapshot);
+    }
+    @Action(name = ACTION_GET_SNAPSHOT)
+    @Override
+    @Nonnull
+    public Task<DataProcessSnapshot> getSnapshot(@ActionParam(PARAM_URN) @Nonnull String urnString,
+                                                 @ActionParam(PARAM_ASPECTS) @Optional("[]") @Nonnull String[] aspectNames) {
+        return super.getSnapshot(urnString, aspectNames);
+    }
+
+    @Action(name = ACTION_BACKFILL)
+    @Override
+    @Nonnull
+    public Task<String[]> backfill(@ActionParam(PARAM_URN) @Nonnull String urnString,
+                                   @ActionParam(PARAM_ASPECTS) @Optional("[]") @Nonnull String[] aspectNames) {
+        return super.backfill(urnString, aspectNames);
+    }
```

Comment:
Also add the Search finder like this to enable search?https://github.com/linkedin/datahub/blob/master/gms/impl/src/main/java/com/linkedin/metadata/resources/dataset/Datasets.java#L188

Reply:
OK. what's this `search finder` for exactly?

Reply:
Actually never mind. Looks like you haven't implemented the `search` method in your client. We can do that in the next PR.